classes/form/CustomerPersister.php

```diff
@@ -188,6 +188,19 @@ private function create(Customer $customer, $clearTextPassword)
             return false;
         }
 
+        /**
+         * Database optimization
+         * We check if the email has been assigned to guest account
+         */
+        $check = Customer::customerExists($customer->email, true, false);
+        if ($check) {
+            $customer = new Customer($check);
```

Comment:
You should use a more explicit name for the data you store in this variable.

For instance:

```php
        $customerId = Customer::customerExists($customer->email, true, false);
        if ($customerId) {
            $customer = new Customer($customerId);
        }
```